src/librbd/crypto/FormatRequest.cc

```diff
@@ -99,6 +101,64 @@ void FormatRequest<I>::handle_flush(int r) {
   if (r != 0) {
     lderr(m_image_ctx->cct) << "unable to flush image: " << cpp_strerror(r)
                             << dendl;
+    finish(r);
+    return;
+  }
+
+  if (m_image_ctx->parent != nullptr) {
+    metadata_set();
+  } else {
+    finish(0);
+  }
+}
+
+template <typename I>
+void FormatRequest<I>::metadata_set() {
+  // serialize parent cryptor
+  auto parent_crypto = m_image_ctx->parent->get_crypto();
+  if (parent_crypto == nullptr) {
+    m_serialized_parent_cryptor.append(
```

Comment:
I would actually recommend re-using `SavedCryptoInterface` but add an enum to it to indicate the state. Keeps the format consistent for decode and better supports changes in the future.

Reply:
I will use an empty key to denote this state.